morepath/request.py

```diff
@@ -138,22 +135,16 @@ def view(self, obj, default=None, app=SAME_APP, **predicates):
         predicates['model'] = obj.__class__
 
         def find(app, obj):
-            return generic.view.component_key_dict(lookup=app.lookup,
-                                                   **predicates)
+            return app.get_view.component_key_dict(**predicates)
 
         view, app = app._follow_defers(find, obj)
         if view is None:
             return default
 
         old_app = self.app
-        old_lookup = self.lookup
-        app.set_implicit()
         self.app = app
-        self.lookup = app.lookup
         result = view.func(obj, self)
```

Comment:
I was going to ask you to explain to me how the ``func`` attribute of the ``View`` object can be accessed in the face of a possible wrapper object? On my branch I had to create a workaround, i.e. ``view.value`` that gives the unwrapped object. The answer is that you register the view unwrapped as opposed to be wrapped. That's interesting and I recall for some reason I couldn't do that -- I need to recheck that.